src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -12,13 +22,114 @@
 from mozanalysis.sizing import HistoricalTarget
 from mozanalysis.utils import get_time_intervals
 
-from scipy.stats import norm
-from math import pi
-from statsmodels.stats.power import tt_ind_solve_power, zt_ind_solve_power
-from statsmodels.stats.proportion import samplesize_proportions_2indep_onetail
-import numpy as np
-import pandas as pd
-import matplotlib.pyplot as plt
+
+class SampleSizeResultsHolder(UserDict):
+    """
+    Object to hold results from different methods.  It extends
+    the dictionary objects so that users can interact with it
+    with a dictionary with the same keys/values as before, making
+    it backward compatible.  The dictionary functionality is extended to include
+    additional attributes to hold metadata, a method for plotting results and
+    a method for returning results as a dataframe
+    """
+    def __init__(self, *args,
+                 metrics: dict = None,
+                 params: dict = None,
+                 **kwargs):
+        """
+        Args:
+            metrics (dict, optional): _description_. Defaults to None.
+            params (dict, optional): _description_. Defaults to None.
+        """
+        super().__init__(*args, **kwargs)
+        self._metrics = metrics
+        self._params = params
+        # create this attribute to hold only the results data
+        # this dict is what was returned from the sample size
+        # methods historically
+        self.results = {k: v for k, v in self.data.items()
+                        if k not in ['metrics', 'params']}
```

Comment:
What about keeping this in `data` rather than creating a separate `results` dict, ie. modifying `data`? That way all the dunder methods like `__getitem__` work out of the box without needing to be overridden.

Reply:
Good catch, fix in most recent commit